.gitattributes

```diff
@@ -1,2 +1,13 @@
+* text=auto
```

Comment:
I used `text=auto` for its simplicity. We are testing with it on Windows now, so we know it's most likely working fine. But I agree it can be a pain to debug in extreme cases, I'll try to make something along the lines you suggested.